doc/source/whatsnew/v1.0.1.rst

```diff
@@ -16,6 +16,8 @@ including other versions of pandas.
 Bug fixes
 ~~~~~~~~~
 
+- Bug in :meth:`AbstractHolidayCalendar.holidays` when no rules were defined (:issue:`31415`)
```

Comment:
Is this bug new between 0.25.x and 1.0.0? If not should target 1.1.0 instead of 1.0.1

Reply:
It looks like this bug has been present for the last 6 years. I don't think the `AbstractHolidayCalendar` functionality gets all that much use and this is a particular corner case.

Given that this PR fixes a bug and changes no apis or behaviours, other than returning a sensible value instead of throwing an unintended exception, I think this should get in the soonest possible release which IIUC would be `1.0.1`.

I'm not sure why *when* the bug was introduced should have any bearing on when it can be fixed. I think it's fine to fix bugs which don't change any apis or behaviours in patch releases - I think this is exactly what patch releases are for!